internal/exec/engine.go

```diff
@@ -162,11 +163,30 @@ func logStructuredJournalEntry(config string, src string, isReferenced bool) err
 	return nil
 }
 
-// acquireConfig returns the configuration, first checking a local cache
-// before attempting to fetch it from the provider.
-func (e *Engine) acquireConfig() (cfg types.Config, err error) {
+// acquireConfig returns the configuration from a local cache. If one
+// does not exist then it will attempt to fetch the configuration from
+// the provider when in fetch stages.
+func (e *Engine) acquireConfig(stageName string) (cfg types.Config, err error) {
+	cfg, err = e.acquireCachedConfig()
+	if err != nil {
+		if os.IsNotExist(err) && strings.HasPrefix(stageName, "fetch") {
+			e.Logger.Info("fetching provider config")
+			cfg, err = e.acquireProviderConfig()
+			if err == errors.ErrEmpty {
+				e.Logger.Info("%v: provider config was empty, writing empty cache config", err)
+				if err = renameio.WriteFile(e.ConfigCache, []byte("{}"), 0640); err != nil {
+					e.Logger.Crit("failed to empty write cached config: %v", err)
+				}
+			}
+			return
+		}
+	}
+	return
```

Comment:
Was expecting a `return err` here, but then saw the signature. The "named return arguments" thing that Golang has really trips me up sometimes.